tools/list_artifacts.py

```diff
@@ -0,0 +1,327 @@
+          yield from self._GenerateWMIQuery(name, doc, source)
+
+  def PrintArtifacts(self, show_docs=False):
+    """Prints artifacts in a key-value list."""
+    for artifact in sorted(self._LoadArtifacts(),
+                           key=lambda artifact: artifact['name']):
+      print('Name:       ', artifact['name'])
+      print('Type:       ', artifact['type'])
+      print('OS:         ', artifact['os'])
+      if 'path' in artifact:
+        print('Path:       ', artifact['path'])
+      if 'key' in artifact:
+        print('Key:        ', artifact['key'])
+      if 'value' in artifact:
+        print('Value:      ', artifact['value'])
+      if 'query' in artifact:
+        print('Query:      ', artifact['query'])
+      if 'base_object' in artifact:
+        print('Base Object:', artifact['base_object'])
+      if 'cmd' in artifact:
+        print('Command:    ', artifact['cmd'])
+      if 'args' in artifact:
+        print('Arguments:  ', artifact['args'])
+      if show_docs:
+        print('Docs:       ', artifact['doc'])
+      print()
+
+  def PrintJSONL(self):
+    """Prints artifacts in a JSON-L format."""
+    for artifact in self._LoadArtifacts():
+      print(artifact)
+
+  def PrintCSV(self):
+    """Prints artifacts in a CSV format."""
+    fieldnames = [
+        'name', 'type', 'os', 'path', 'key', 'value', 'names', 'query',
+        'base_object', 'cmd', 'args', 'doc']
+    writer = csv.DictWriter(sys.stdout, fieldnames=fieldnames)
+    writer.writeheader()
+
+    for artifact in self._LoadArtifacts():
+      writer.writerow(artifact)
+
+
+def Main():
+  """The main program function.
+
+  Returns:
+    bool: True if successful or False if not.
+  """
+  args_parser = argparse.ArgumentParser(
+    description='List artifact definitions.')
+
+  args_parser.add_argument(
+      'definitions', nargs='?', action='store', metavar='PATH',
+      help=('path of the file or directory that contains the artifact '
+            'definitions.'),
+      default='./data')
+
+  args_parser.add_argument(
+      'os', choices=definitions.SUPPORTED_OS,
+      help='show artifacts from operating system.')
+
+  args_parser.add_argument(
+      '--format',
+      choices=['list', 'jsonl', 'csv'], default='list',
+      help='the output format.')
+
+  args_parser.add_argument('--show_docs', action='store_true',
+      help='show docs value (only for list output format).')
+
+  args_parser.add_argument('--split_paths', action='store_true',
+      help='show separate entries for artifacts with more than one path.')
+
+  options = args_parser.parse_args()
+
+  if not options.definitions:
+    print('Source value is missing.')
+    print('')
+    args_parser.print_help()
+    print('')
+    return False
+
+  if not os.path.exists(options.definitions):
+    print(f'No such file or directory: {options.definitions}')
+    print('')
+    return False
+
+  artifact_lister = ArtifactLister(
```

Comment:
why not use ArtifactDefinitionsRegistry ? given that you want to make sure you have the necessary definitions for groups.